app/models/appointment.rb

```diff
@@ -1,5 +1,23 @@
 class Appointment < ActiveRecord::Base
 	belongs_to :doctor
 	belongs_to :patient
-	validates_presence_of :doctor
+
+  def self.overload(doctor_id,schedule)
+    appointment_count = Appointment.where(doctor_id: doctor_id, status:'Approved', schedule: schedule ).count
```

Comment:
we could have a scope approved_appointments on the Appointment class, and we could then chain that scope on Activerecord relations. Something like `Appointment.where(doctor_id: doctor_id, schedule: schedule ).approved_appointments.count > SOME_CONST`.